packages/pricing/src/services/price-set-money-amount-rules.ts

```diff
@@ -15,96 +9,54 @@ type InjectedDependencies = {
 
 export default class PriceSetMoneyAmountRulesService<
   TEntity extends PriceSetMoneyAmountRules = PriceSetMoneyAmountRules
-> {
-  protected readonly priceSetMoneyAmountRulesRepository_: DAL.RepositoryService
+> extends ModulesSdkUtils.abstractServiceFactory<
+  InjectedDependencies,
+  {
+    create: ServiceTypes.CreatePriceSetMoneyAmountRulesDTO
+    update: ServiceTypes.UpdatePriceSetMoneyAmountRulesDTO
+  }
+>(PriceSetMoneyAmountRules)<TEntity> {
+  protected readonly priceSetMoneyAmountRulesRepository_: DAL.RepositoryService<TEntity>
 
   constructor({ priceSetMoneyAmountRulesRepository }: InjectedDependencies) {
+    // @ts-ignore
+    super(...arguments)
     this.priceSetMoneyAmountRulesRepository_ =
       priceSetMoneyAmountRulesRepository
   }
 
   @InjectManager("priceSetMoneyAmountRulesRepository_")
-  async retrieve(
-    priceSetMoneyAmountRulesId: string,
-    config: FindConfig<ServiceTypes.PriceSetMoneyAmountRulesDTO> = {},
-    @MedusaContext() sharedContext: Context = {}
-  ): Promise<TEntity> {
-    return (await retrieveEntity<
-      PriceSetMoneyAmountRules,
-      ServiceTypes.PriceSetMoneyAmountRulesDTO
-    >({
-      id: priceSetMoneyAmountRulesId,
-      identifierColumn: "id",
-      entityName: PriceSetMoneyAmountRules.name,
-      repository: this.priceSetMoneyAmountRulesRepository_,
-      config,
-      sharedContext,
-    })) as TEntity
-  }
-
-  @InjectManager("priceSetMoneyAmountRulesRepository_")
-  async list(
+  async list<TEntityMethod = ServiceTypes.PriceSetMoneyAmountRulesDTO>(
     filters: ServiceTypes.FilterablePriceSetMoneyAmountRulesProps = {},
-    config: FindConfig<ServiceTypes.PriceSetMoneyAmountRulesDTO> = {},
+    config: FindConfig<TEntityMethod> = {},
     @MedusaContext() sharedContext: Context = {}
   ): Promise<TEntity[]> {
-    return (await this.priceSetMoneyAmountRulesRepository_.find(
+    return await this.priceSetMoneyAmountRulesRepository_.find(
       this.buildQueryForList(filters, config),
       sharedContext
-    )) as TEntity[]
+    )
   }
 
   @InjectManager("priceSetMoneyAmountRulesRepository_")
-  async listAndCount(
+  async listAndCount<TEntityMethod = ServiceTypes.PriceSetMoneyAmountRulesDTO>(
     filters: ServiceTypes.FilterablePriceSetMoneyAmountRulesProps = {},
-    config: FindConfig<ServiceTypes.PriceSetMoneyAmountRulesDTO> = {},
+    config: FindConfig<TEntityMethod> = {},
     @MedusaContext() sharedContext: Context = {}
   ): Promise<[TEntity[], number]> {
-    return (await this.priceSetMoneyAmountRulesRepository_.findAndCount(
+    return await this.priceSetMoneyAmountRulesRepository_.findAndCount(
       this.buildQueryForList(filters, config),
       sharedContext
-    )) as [TEntity[], number]
+    )
   }
 
-  private buildQueryForList(
+  private buildQueryForList<
```

Comment:
question(nit): for these services where we've used a private `buildQueryForList` but only call `ModuleSdkUtils.buildQuery` it seem to me that we should still be able to use the abstract class method since it's the same but with more steps?

Reply:
Yes i will change that, oli did some partial changes but this is something i ll finish tomorrow ☺️